### PR TITLE
WiFi and network usb requests

### PR DESCRIPTION
**Pull request by avtolstoy**
## Review only. Superseded by #1444

<details>
  <summary><i>submission notes</i></summary>

```
**Important:** Please sanitize/remove any confidential info like usernames, passwords, org names, product names/ids, access tokens, client ids/secrets, or anything else you don't wish to share.

Please Read and Sign the Contributor License Agreement ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md)).

You may also delete this submission notes header if you'd like. Thank you for contributing!
```
</details>

### Problem

This PR implements the following usb requests:
- Set active WiFi antenna: `CTRL_REQUEST_WIFI_SET_ANTENNA`
- Get active WiFi antenna: `CTRL_REQUEST_WIFI_GET_ANTENNA`
- Scan WiFi networks: `CTRL_REQUEST_WIFI_SCAN`
- Add WiFi access point configuration: `CTRL_REQUEST_WIFI_SET_CREDENTIALS`
- List configured access points: `CTRL_REQUEST_WIFI_GET_CREDENTIALS`
- Clear all configured access points: `CTRL_REQUEST_WIFI_CLEAR_CREDENTIALS`
- Change network interface configuration: `CTRL_REQUEST_NETWORK_SET_CONFIGURATION`
- Get network interface configuration: `CTRL_REQUEST_NETWORK_GET_CONFIGURATION`
- Retrieve network interface status and active configuration: `CTRL_REQUEST_NETWORK_GET_STATUS`

Protocol files: [wifi.proto](https://github.com/spark/firmware/blob/248a952a7f2cd29224c509919f8e3c23e71169bd/system/src/control/wifi.proto) [network.proto](https://github.com/spark/firmware/blob/248a952a7f2cd29224c509919f8e3c23e71169bd/system/src/control/network.proto)

### Steps to Test

N/A

### Example App

N/A

### References

- [CH9462]
- [CH9463]

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: hal/inc/hal_dynalib_wlan.h [hunk 1]
```diff
@@ -78,6 +78,8 @@ DYNALIB_FN(27, hal_wlan, wlan_set_hostname, int(const char*, void*))
 DYNALIB_FN(28, hal_wlan, wlan_get_hostname, int(char*, size_t, void*))
 DYNALIB_FN(29, hal_wlan, wlan_connected_info, int(void*, wlan_connected_info_t*, void*))
 DYNALIB_FN(30, hal_wlan, wlan_get_antenna, WLanSelectAntenna_TypeDef(void*))
+DYNALIB_FN(31, hal_wlan, wlan_get_ipaddress, int(IPConfig*, void*))
```
Comment (m-mcgowan) [inline hal/inc/hal_dynalib_wlan.h]: how about we combine this into a more generic get IP configuration?

File: system/src/system_control_internal.cpp [hunk 2]
```diff
@@ -258,7 +364,279 @@ void particle::SystemControl::processRequest(ctrl_request* req, ControlRequestCh
         setResult(req, r);
         break;
     }
+    case CTRL_REQUEST_WIFI_SCAN:
```
Comment (m-mcgowan) [inline system/src/system_control_internal.cpp]: I would prefer these handlers to be broken out into separate functions. It makes reading a lot easier and the compilers job easier too, leading to potentially smaller code.

Comment (avtolstoy) [reply system/src/system_control_internal.cpp]: These are already refactored in #1444. See https://github.com/particle-iot/firmware/tree/a3b09a11bf71cbf9cacb8961578ca9de55bb14d6/system/src/control

Comment (m-mcgowan) [reply system/src/system_control_internal.cpp]: sweet!

File: hal/inc/hal_dynalib_wlan.h (hunk 1 above)
Comment (avtolstoy) [reply hal/inc/hal_dynalib_wlan.h]: It does actually retrieve the full *saved* configuration. The idea was for this method to be analogous to `wlan_set_ipaddress, void(const HAL_IPAddress*, const HAL_IPAddress*, const HAL_IPAddress*, const HAL_IPAddress*, const HAL_IPAddress*, void*)`, but I can see how something like `wlan_get_stored_configuration()` would work better. Let me know your thoughts on naming.

**Pull request closed without merging**